### ISSUE TITLE
Allow plugabble uglifyjs config

### ISSUE THREAD

**Issue by graingert** [webpack-contrib/uglifyjs-webpack-plugin#17]
I'd like to use https://www.npmjs.com/package/uglify-js-harmony rather than the git branch of uglify-js, it would be handy if the plugin config supported:
```js
import uglify from 'uglify-js-harmony';

ujsp = new UglifyJSPlugin({uglify});
```

**Comment by bebraw** [comment]
Can you find that plugin on GitHub? Maybe doing something like `"uglify-js": "... path to custom github version ..."` would work. That's what I would try first.

If that doesn't work, I guess we could accept a PR although doing that would make the package out of sync with webpack core unless it's accepted there too.

Babili is a worthwhile option by the way.

**Comment by graingert** [comment]
Yeah I'm trying to avoid github URLs because it's not supported in shrinkwrap

**Comment by bebraw** [comment]
@d3viant0ne Any thoughts on this?

**Comment by graingert** [comment]
@hamosapience ^

**Comment by d3viant0ne** [comment]
My initial concerns would be the side effects to supporting UglifyJS 2 in this loader side by side with UgilfyJS. I'd need to study the changes between standard & harmony before making a determination on the above.

Second concern is related to the thread about the future of uglify and possible merge with Babels babili. This has the potential to increase the support cost & complexity of the plugin for something who's lifespan is in question.

Point being, adding the ability to use means we have to support issue's that arise from it's inclusion and there is a great deal of uncertainty around the future plans for Uglify2 last I checked.

**Comment by bebraw** [comment]
Alright.

@graingert Here's what I suggest. Fork the webpack plugin and publish under `@graingert/uglifyjs-webpack-plugin`. Scopes are free for personal accounts. This also avoids the maintenance cost issue Joshua brought up.

Would that work?

[babili-webpack-plugin](https://github.com/webpack-contrib/babili-webpack-plugin) is starting to look good by the way. Don't mind the version number. It's in sync with Babili itself.

**Comment by michael-ciniawsky** [comment]
Will be 'fixed' by #63
